### PR TITLE
Support OAuth2 login for the UI Apps

### PR DESCRIPTION
**Pull request by filiphr**
Additionally add Keycloak based IdmIdentityService in the UI IDM App

### CODE REVIEW

File: modules/flowable-ui/flowable-ui-idm-logic/src/main/java/org/flowable/ui/idm/service/keycloak/KeycloakGroupQueryImpl.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.flowable.common.engine.api.FlowableException;
+import org.flowable.common.engine.impl.interceptor.CommandContext;
+import org.flowable.idm.api.Group;
+import org.flowable.idm.api.User;
```
Comment (dbmalkovsky) [inline modules/flowable-ui/flowable-ui-idm-logic/src/main/java/org/flowable/ui/idm/service/keycloak/KeycloakGroupQueryImpl.java:24]: I believe this import is not used.

File: modules/flowable-ui/flowable-ui-idm-logic/src/main/java/org/flowable/ui/idm/service/keycloak/KeycloakGroupQueryImpl.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+import org.flowable.idm.engine.impl.GroupQueryImpl;
+import org.flowable.idm.engine.impl.persistence.entity.GroupEntityImpl;
+import org.flowable.idm.engine.impl.persistence.entity.UserEntityImpl;
```
Comment (dbmalkovsky) [inline modules/flowable-ui/flowable-ui-idm-logic/src/main/java/org/flowable/ui/idm/service/keycloak/KeycloakGroupQueryImpl.java:27]: As is this one also not used.

File: modules/flowable-ui/flowable-ui-idm-rest/src/main/java/org/flowable/ui/idm/rest/app/IdmProfileResource.java [hunk 3]
```diff
@@ -27,8 +27,10 @@
 import org.flowable.ui.common.service.exception.InternalServerErrorException;
 import org.flowable.ui.common.service.exception.NotFoundException;
 import org.flowable.ui.idm.model.ChangePasswordRepresentation;
+import org.flowable.ui.idm.model.UserInformation;
```
Comment (dbmalkovsky) [inline modules/flowable-ui/flowable-ui-idm-rest/src/main/java/org/flowable/ui/idm/rest/app/IdmProfileResource.java:30]: The import of `org.flowable.idm.api.User` at line 23 is no longer used.

File: modules/flowable-ui/flowable-ui-modeler-logic/src/main/java/org/flowable/ui/modeler/service/FlowableDecisionTableService.java [hunk 4]
```diff
@@ -35,8 +35,8 @@
 import org.flowable.dmn.editor.converter.DmnJsonConverter;
 import org.flowable.dmn.model.DmnDefinition;
 import org.flowable.dmn.xml.converter.DmnXMLConverter;
-import org.flowable.idm.api.User;
 import org.flowable.ui.common.model.ResultListDataRepresentation;
+import org.flowable.ui.common.security.SecurityScope;
 import org.flowable.ui.common.security.SecurityUtils;
```
Comment (dbmalkovsky) [inline modules/flowable-ui/flowable-ui-modeler-logic/src/main/java/org/flowable/ui/modeler/service/FlowableDecisionTableService.java:40]: I don't believe this import is needed either.

File: modules/flowable-ui/flowable-ui-modeler-rest/src/main/java/org/flowable/ui/modeler/rest/api/ApiModelResource.java [hunk 5]
```diff
@@ -17,7 +17,7 @@
 import org.apache.commons.lang3.StringUtils;
 import org.flowable.bpmn.converter.BpmnXMLConverter;
 import org.flowable.editor.language.json.converter.BpmnJsonConverter;
-import org.flowable.idm.api.User;
+import org.flowable.ui.common.security.SecurityScope;
```
Comment (dbmalkovsky) [inline modules/flowable-ui/flowable-ui-modeler-rest/src/main/java/org/flowable/ui/modeler/rest/api/ApiModelResource.java:20]: Again `SecurityScope` is not needed I believe.

File: modules/flowable-ui/flowable-ui-task-logic/src/main/java/org/flowable/ui/task/service/runtime/FlowableCaseInstanceService.java [hunk 6]
```diff
@@ -35,6 +35,7 @@
 import org.flowable.form.api.FormService;
 import org.flowable.idm.api.User;
 import org.flowable.ui.common.model.ResultListDataRepresentation;
+import org.flowable.ui.common.security.SecurityScope;
```
Comment (dbmalkovsky) [inline modules/flowable-ui/flowable-ui-task-logic/src/main/java/org/flowable/ui/task/service/runtime/FlowableCaseInstanceService.java:38]: This looks like the third and last case where `SecurityScope` is not needed.

**Comment by filiphr** [comment]
Thanks @dbmalkovsky, I'll apply the feedback

**Pull request merged**